fdbserver/DataDistribution.actor.cpp

```diff
@@ -997,6 +1009,23 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 		return Void();
 	}
 
+	// Check if server or machine has a valid locality based on configured replication policy
+	bool isValidLocality(Reference<IReplicationPolicy> storagePolicy, LocalityData& locality) {
+		if (!SERVER_KNOBS->DD_VALIDATE_LOCALITY) {
```

Comment:
Is the idea that we'll get rid of this in 7.0 or whenever we add a test that triggers the failure case?

Reply:
This is just in case the change broke anything, we can use the knob to disable the change.

Reply:
When we do add a test case to trigger the broken behavior (and I hope we eventually do), simulation is going to fail when we buggify into here. Maybe we should add a comment to help save our future selves some time (perhaps on the assert that will fail?).

Reply:
Good idea. I will add a comment there.